### ISSUE TITLE
Selenium expects HTTP status 400 on ‘no such alert’ error

### ISSUE THREAD

**Issue by adam-zjy** [SeleniumHQ/selenium#4815]
## System
* GeckoDriver Version: 0.19.0
* Platform: Windows/MacOS
* Firefox: 56
* Selenium: 3.5.3

## Testcase

We want to verify an alert is presented on the page after the page load, when call the switch().alert() it will return null immediately rather than wait for the page loading completely

## Trace-level log

```
1506663782880	webdriver::server	DEBUG	-> GET /session/eca71e6a-a826-48cf-a83f-720ec958c490/alert/text 
1506663782881	geckodriver::marionette	TRACE	-> 29:[0,73,"getTextFromDialog",{}]
1506663782883	Marionette	TRACE	0 -> [0,73,"getTextFromDialog",{}]
1506663782885	Marionette	TRACE	0 <- [1,73,{"error":"no such alert","message":"No modal dialog is currently open","stacktrace":"WebDriverError@chrome://marionette/content/error.js:227:5\nNoAlertOpenError@chrome://marionette/content/error.js:448:5\nGeckoDriver.prototype._checkIfAlertIsPresent@chrome://marionette/content/driver.js:3088:11\nGeckoDriver.prototype.getTextFromDialog@chrome://marionette/content/driver.js:3043:3\nexecute/req<@chrome://marionette/content/server.js:539:22\nTaskImpl_run@resource://gre/modules/Task.jsm:331:42\nTaskImpl@resource://gre/modules/Task.jsm:280:3\nasyncFunction@resource://gre/modules/Task.jsm:252:14\nTask_spawn@resource://gre/modules/Task.jsm:166:12\nexecute@chrome://marionette/content/server.js:529:15\nonPacket@chrome://marionette/content/server.js:500:7\n_onJSONObjectReady/<@chrome://marionette/content/transport.js:501:9\n"},null]
1506663782885	geckodriver::marionette	TRACE	<- [1,73,{"error":"no such alert","message":"No modal dialog is currently open","stacktrace":"WebDriverError@chrome://marionette/content/error.js:227:5\nNoAlertOpenError@chrome://marionette/content/error.js:448:5\nGeckoDriver.prototype._checkIfAlertIsPresent@chrome://marionette/content/driver.js:3088:11\nGeckoDriver.prototype.getTextFromDialog@chrome://marionette/content/driver.js:3043:3\nexecute/req<@chrome://marionette/content/server.js:539:22\nTaskImpl_run@resource://gre/modules/Task.jsm:331:42\nTaskImpl@resource://gre/modules/Task.jsm:280:3\nasyncFunction@resource://gre/modules/Task.jsm:252:14\nTask_spawn@resource://gre/modules/Task.jsm:166:12\nexecute@chrome://marionette/content/server.js:529:15\nonPacket@chrome://marionette/content/server.js:500:7\n_onJSONObjectReady/<@chrome://marionette/content/transport.js:501:9\n"},null]
1506663782887	webdriver::server	DEBUG	<- 404 Not Found {"value":{"error":"no such alert","message":"No modal dialog is currently open","stacktrace":"stack backtrace:\n   0:           0x4821c4 - <no info>\n   1:           0x482933 - <no info>\n   2:           0x445159 - <no info>\n   3:           0x44eff6 - <no info>\n   4:           0x4322e1 - <no info>\n   5:           0x40ba4e - <no info>\n   6:           0x4119b9 - <no info>\n   7:           0x6d2c89 - <no info>\n   8:           0x426006 - <no info>\n   9:           0x6cd490 - <no info>\n  10:     0x7ffc8f738102 - BaseThreadInitThunk"}}
Sep 29, 2017 1:43:03 PM org.openqa.selenium.remote.ErrorCodes toStatus
INFO: HTTP Status: '404' -> incorrect JSON status mapping for 'no such alert' (400 expected)
```

**Comment by adam-zjy** [comment]
This issue is moved from https://github.com/mozilla/geckodriver/issues/980

**Comment by barancev** [comment]
Fixed by 0863b2246d3c6c9df31c43ccad643d6fef985728

But it's just an info message that does not affect error handling. I'm sure this change will not resolve your original issue, whatever it is.
